tests/src/test/java/org/eclipse/hono/tests/registry/TenantApiTests.java

```diff
@@ -83,14 +83,14 @@ public void testGetTenant(final VertxTestContext ctx) {
                         .put("deployment", new JsonObject()
                                 .put("maxInstances", 4)));
         final String tenantId = getHelper().getRandomTenantId();
-        final TenantObject tenant = TenantObject.from(tenantId, true)
-                .setResourceLimits(resourceLimits)
-                .setDefaults(defaults)
-                .setAdapterConfigurations(adapterConfig)
-                .setProperty("customer", "ACME Inc.");
+        final JsonObject tenantPayload = TenantUtil.newTenant(tenantId, true)
+                .put(TenantConstants.FIELD_RESOURCE_LIMITS, resourceLimits)
+                .put(TenantConstants.FIELD_PAYLOAD_DEFAULTS, defaults)
+                .put(TenantConstants.FIELD_ADAPTERS, adapterConfig)
+                .put("customer", "ACME Inc.");
```

Comment:
this kind of code where we need to use the property names explicitly is exactly what we want to prevent by using a value object.
FMPOV we first need to create a new value object to use for interacting with the Management API (e.g. TenantManagementObject`?) and use it in all code where we use the HTTP API to e.g. register a tenant in the integration tests. `TenantUtil` doesn't really serve that purpose FMPOV.
We can then remove everything from `TenantObject` that is not needed.

Reply:
see `org.eclipse.hono.service.management.tenant.Tenant`

Reply:
yes sure 👍 I have seen that value object class

Reply:
I don't get it. Why is that a reason not to use the `Tenant` class? We only need it for setting up the  fixture, right? We do so utilizing the `DeviceRegistryHttpClient`, so all we would need to do is have that client also support passing in a `Tenant` object instead of a `JsonObject`, right?

Reply:
sure.. am working on this now

Reply:
I updated this PR. the build passes on my local machine.. don't know why its failing on travis